### PR TITLE
Added a widget for custom maps

### PR DESCRIPTION
**Pull request by eskyuu**
This adds a widget for custom maps.  They should auto-scale to fit the widget window.

DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [ ] Have you followed our [code guidelines?](https://docs.librenms.org/Developing/Code-Guidelines/)
- [ ] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.
- [ ] If my Pull Request makes discovery/polling/yaml changes, I have added/updated [test data](https://docs.librenms.org/Developing/os/Test-Units/).

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

**Comment by murrant** [comment]
Nice!  We probably want to put some of the JS in a place to share it so it isn't duplicated.

**Comment by eskyuu** [comment]
How does this look to you?

I looked at the way the custom map modals are imported directly into the content section, and figured this was the best way to do partial imports.

For the widget I am importing into the default overview page, otherwise we would load the same functions times if more than 1 custom map widget was put on the same page.  I figure it is more targeted than putting it all in librenms.js.

**Review by laf — DISMISSED** [review]
LGTM, tested as well.

File: resources/views/map/custom-js.blade.php [hunk 1]
```diff
@@ -0,0 +1,182 @@
+<script type="text/javascript" src="{{ asset('js/vis.min.js') }}"></script>
+<script type="text/javascript">
+    function custommapLegendPctDefaultColour(pct) {
```
Comment (murrant) [inline resources/views/map/custom-js.blade.php:3]: This is almost enough functions to put them under their own namespace.  It is fine as is, just suggesting now as it is harder to change later.

Comment (eskyuu) [reply resources/views/map/custom-js.blade.php]: I've updated to use a namespace.  Does the revised code look good, or is the name too generic, or is there a different preferred way to initialise the namespace?

Comment (murrant) [reply resources/views/map/custom-js.blade.php]: Sorry, JS is definitely not my strong suit :)  Can't answer those questions. The name is good.

**Review by laf — DISMISSED** [review]
Haven't code reviewed but tested and works fine.

**Comment by eskyuu** [comment]
Had to fix a merge conflict

**Comment by eskyuu** [comment]
@laf Can you please re-approve this.  I had to fix a conflict with the js refactor.  This is also stopping me from submitting a couple more PRs for custom maps, which build off the changes made here.  Who can do the code review so it can be merged?

**Review by laf — APPROVED** [review]
Tested and all works fine here.

**Comment by eskyuu** [comment]
Thanks

**Pull request merged**